core/src/validator.rs

```diff
@@ -596,25 +598,8 @@ impl Validator {
             ));
         }
         let genesis_config = load_genesis(config, ledger_path)?;
-
         metrics_config_sanity_check(genesis_config.cluster_type)?;
 
-        if let Some(expected_shred_version) = config.expected_shred_version {
-            if let Some(wait_for_supermajority_slot) = config.wait_for_supermajority {
-                *start_progress.write().unwrap() = ValidatorStartProgress::CleaningBlockStore;
-                backup_and_clear_blockstore(
-                    ledger_path,
-                    config,
-                    wait_for_supermajority_slot + 1,
```

Comment:
One kind of obscure corner case here is that the old code would perform the scan with wait-for-supermajority in the future (say slot `999_999_999`). With my new impl, we do NOT do that.

Failing to remove wfsm from validator args is just forgetfulness, but setting an invalid valid is true operator error so I'm not too concerned about this case. Just mentioning for completeness tho